source/base/timer.cc

```diff
@@ -353,8 +353,13 @@ TimerOutput::TimerOutput (MPI_Comm      mpi_communicator,
 
 TimerOutput::~TimerOutput()
 {
-  while (active_sections.size() > 0)
-    leave_subsection();
+  try
+    {
+      while (active_sections.size() > 0)
+        leave_subsection();
```

Comment:
Can we at least terminate the program, rather than ignoring the exception in the catch clause? It would seem to me that nothing positive can come out of continuing here.